src/application_input.hpp

```diff
@@ -16,19 +16,77 @@
 #include <functional>
 #include <map>
 #include <memory>
+#include <sstream>
 #include <string>
+#include <unordered_map>
 #include <vector>
 
 #include "bvals/boundary_conditions.hpp"
 #include "defs.hpp"
 #include "interface/state_descriptor.hpp"
+#include "interface/swarm_boundaries.hpp"
+#include "kokkos_abstraction.hpp"
 #include "parameter_input.hpp"
 #include "parthenon_arrays.hpp"
+#include "utils/error_checking.hpp"
 
 namespace parthenon {
 
-struct ApplicationInput {
+class ApplicationInput {
  public:
+  ApplicationInput() {
+    RegisterBoundaryCondition(BoundaryFace::inner_x1, "outflow",
+                              &BoundaryFunction::OutflowInnerX1);
+    RegisterBoundaryCondition(BoundaryFace::outer_x1, "outflow",
+                              &BoundaryFunction::OutflowOuterX1);
+    RegisterBoundaryCondition(BoundaryFace::inner_x2, "outflow",
+                              &BoundaryFunction::OutflowInnerX2);
+    RegisterBoundaryCondition(BoundaryFace::outer_x2, "outflow",
+                              &BoundaryFunction::OutflowOuterX2);
+    RegisterBoundaryCondition(BoundaryFace::inner_x3, "outflow",
+                              &BoundaryFunction::OutflowInnerX3);
+    RegisterBoundaryCondition(BoundaryFace::outer_x3, "outflow",
+                              &BoundaryFunction::OutflowOuterX3);
+    RegisterBoundaryCondition(BoundaryFace::inner_x1, "reflecting",
+                              &BoundaryFunction::OutflowInnerX1);
+    RegisterBoundaryCondition(BoundaryFace::outer_x1, "reflecting",
+                              &BoundaryFunction::ReflectOuterX1);
+    RegisterBoundaryCondition(BoundaryFace::inner_x2, "reflecting",
+                              &BoundaryFunction::ReflectInnerX2);
+    RegisterBoundaryCondition(BoundaryFace::outer_x2, "reflecting",
+                              &BoundaryFunction::ReflectOuterX2);
+    RegisterBoundaryCondition(BoundaryFace::inner_x3, "reflecting",
+                              &BoundaryFunction::ReflectInnerX3);
+    RegisterBoundaryCondition(BoundaryFace::outer_x3, "reflecting",
+                              &BoundaryFunction::ReflectOuterX3);
```

Comment:
I'm wondering if we should register these at all (or register those in the respective applications that use them).
The reason is that reflecting boundary conditions depend either on properly using "vectors" (with the correct number of dimensions), i.e., with downstream application using an Athen~a~-esque approach will not work out of the box.
Not registering them here would allow downstream codes to either explicitly implement and register them or fail loudly (when a user tries to use them in the input file without knowing the Parthenon internals).

Bonus point: This would be in line with the registered `Swarm` functions.

Reply:
Currently reflecting is the default boundary condition, so that would be a more breaking change. I wouldn't necessarily object to this, though, so long as we still provide ways of adding reflecting and document it.

Would curious on @jdolence and @lroberts36 's opinion on this one, as riot uses reflecting BCs heavily.

Reply:
Do you/we have any input files that reply on implicitly specified default boundary conditions?
While this being a breaking change it may not be that breaking in practices (apart from adding the register functions downstream).

Reply:
Yeah I don't think we ever implicitly assume a boundary condition, so as long as we add the code downstream to register the parthenon provided reflecting boundary conditions, I think this would be fine.

Reply:
:+1: Ok sounds good. I will remove the reflecting then and document how to register it.

Reply:
Reflecting BCs no longer automatically registered.